resolwe/flow/models/utils/duplicate.py

```diff
@@ -240,13 +240,13 @@ def processor(datum):
 
         """
         if inherit_collection:
```

Comment:
Maybe you could just rewrite the first condition to `if inherit_collection and datum.collection`?